app/Models/BaseAlbumImpl.php

```diff
@@ -271,6 +276,17 @@ protected function setSortingAttribute(?PhotoSortingCriterion $sorting): void
 		$this->attributes['sorting_order'] = $sorting?->order;
 	}
 
+	/**
+	 * Provide the policy attributes for said album.
+	 *
+	 * @return AlbumProtectionPolicy|null
+	 */
+	protected function getPoliciesAttribute(): AlbumProtectionPolicy|null
+	{
+		// Provide the policies if the user can edit.
+		return Gate::check(AlbumPolicy::CAN_EDIT, [AbstractAlbum::class, $this]) ? AlbumProtectionPolicy::ofBaseAlbum($this) : null;
+	}
+
```

Comment:
I don't mind to add the "virtual" attribute `policies` to the base album model which wraps all the security attributes of an album. At least this imposes a more structure on the class. Personally, I like this approach.

However, if one looks at `AlbumProtectionPolicy` this DTO looks like
```php
return new AlbumProtectionPolicy(
			is_public: $baseAlbum->is_public,
			is_link_required: $baseAlbum->is_link_required,
			is_nsfw: $baseAlbum->is_nsfw,
			grants_full_photo_access: $baseAlbum->grants_full_photo_access,
			grants_download: $baseAlbum->grants_download,
			is_password_required: $baseAlbum->password !== null && $baseAlbum->password !== '',
		);
```
As one can see, the first five attributes of the DTO are a simple copy of direct attributes of the album. The last one (i.e. `is_password_required`) is a direct copy of the virtual attribute `has_password` which is provided by the method `BaseAlbumImpl::getHasPasswordAttribute`. Hence, in serialized form we have an album object like this
```js
{
	"has_password":,                     // <-----------------+
	"grants_download":,                  // <--------------+  |
	"grants_full_photo_access":,         // <-----------+  |  |
	"is_nsfw":,                          // <--------+  |  |  |
	"is_link_required":,                 // <-----+  |  |  |  |
	"is_public":,                        // <--+  |  |  |  |  |
	"policies": {                        //    |  |  |  |  |  |
		"is_public":,                // <--+  |  |  |  |  |
		"is_link_required":,         // <-----+  |  |  |  |
		"is_nsfw":,                  // <--------+  |  |  |
		"grants_full_photo_access":, // <-----------+  |  |
		"grants_download":,          // <--------------+  |
		"is_password_required":,     // <-----------------+
	},
}
```
I don't mind to group these six attributes as `policy` together. But I am strictly against this redundancy. This will only lead to more confusing code (especially in the frontend), because some frontend code will refer to `album.is_public` other frontend code will refer to `album.policies.is_public` and there is no clear line whether one should use the one or the other approach.

If you intent to keep this change, this is what I would do

 1. Keep the `AlbumProtectionPolicy` class, but please rename the attribute to `policy`, not `policies`
 2. Remove (aka hide) the direct attributes from the model to avoid this redundancy
 3. Always transmit the attribute `policy`, not only if the current user can edit the album. I don't see why how this is related. With the current code all users learn the values of the attributes anyway.
 4. Make the attribute `policy` settable, too, i.e. we need a `setPoliciesAttribute`

Of course, the alternative to completely revert this change is still an option, too. But at the current state, this is only half-baked. Optimally, this entire change (if you intend to keep it), would be on an independent PR as it is already huge enough considering the necessary changes in the frontend.

Reply:
> 4. Make the attribute policy settable, too, i.e. we need a setPoliciesAttribute

Not necessarily, especially if we are going to rework it later with improved per User control of the permissions.
This is currently handled by `App\Actions\Album\SetProtectionPolicy`.

Reply:
> Make the attribute policy settable, too, i.e. we need a setPoliciesAttribute

Done.